### PR TITLE
Update/3.3 rc02

### PR DESCRIPTION
**Pull request by jbout**

### CODE REVIEW

**Comment by Babacooll** [comment]
@jbout LGTM but I still need to test it (before the end of the week)

File: docker/phpfpm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+RUN usermod -G staff www-data
+
+RUN apt-get update && apt-get install -y libpng-dev libjpeg-dev libpq-dev zip unzip sudo wget sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*
+
+RUN docker-php-ext-configure gd --with-png-dir=/usr --with-jpeg-dir=/usr
+RUN docker-php-ext-configure pdo_mysql --with-pdo-mysql=mysqlnd
+RUN docker-php-ext-configure pgsql -with-pgsql=/usr/local/pgsql
+RUN docker-php-ext-configure mysqli --with-mysqli=mysqlnd
+
+RUN yes | pecl install igbinary redis
```
Comment (gitromba) [inline docker/phpfpm/Dockerfile:13]: This runs on error as `libzstd` is required for this Redis version (explicit versioning to the rescue):
`
checking for libzstd files in default path... not found
configure: error: Please reinstall the libzstd distribution
ERROR: /tmp/pear/temp/redis/configure --with-php-config=/usr/local/bin/php-config --enable-redis-igbinary=y --enable-redis-lzf=y --enable-redis-zstd=y' failed
ERROR: Service 'tao_phpfpm' failed to build: The command '/bin/sh -c yes | pecl install igbinary redis' returned a non-zero code: 1
`

File: .gitignore [hunk 2]
```diff
@@ -2,10 +2,10 @@
 .idea/
 .buildpath
 .settings/
+.idea/
```
Comment (gitromba) [inline .gitignore:5]: It's in the 1st line. This could be removed.

File: .gitignore [hunk 3]
```diff
@@ -33,8 +32,20 @@ taoTestTaker/
 taoTests/
 taoWfAdvTest/
 taoWfTest/
-vendor/
-wfAuthoring/
-wfEngine/
 taoOutcomeRds/
 taoOutcomeUi/
+pciSamples/
```
Comment (gitromba) [inline .gitignore:37]: A whitelist would be more safe and maintainable. After a TAO install I have extensions, like `taoDeliverySchedule` which are not on the list.

File: docker/LICENSE [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (gitromba) [inline docker/LICENSE:1]: `MIT` AFAIK is `GNU` compatible. Could it be used in TAO products?

File: docker/LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+Copyright (c) 2019 Jonathan VUILLEMIN
```
Comment (gitromba) [inline docker/LICENSE:3]: Is this ok?

**Comment by gitromba** [comment]
The local dev env is not working with the provided docker-compose file due to a missing dependency as described in the `Dockerfile`. As nowadays this is the standard way to spin up applications locally, IMO should be fixed. The fix is straightforward and requires minimal effort. 

Using TAO docker-stack, the application install was successful, I was able to login and to perform a quick smoke test. I've faced **no** visible error or defects upon.

@Babacooll, I advise a more thorough test conducted by QA, I could test it only superficially.

File: docker/LICENSE (hunk 4 above)
Comment (jbout) [reply docker/LICENSE:1]: MIT in GPL2 is fine, see https://en.wikipedia.org/wiki/License_compatibility#Compatibility_of_FOSS_licenses

File: docker/phpfpm/Dockerfile (hunk 1 above)
Comment (Babacooll) [reply docker/phpfpm/Dockerfile:13]: @jbout should be fixed into develop later

File: .gitignore (hunk 2 above)
Comment (Babacooll) [reply .gitignore:5]: @jbout should be fixed into develop later

File: .gitignore (hunk 3 above)
Comment (Babacooll) [reply .gitignore:37]: @jbout should be fixed into develop later

File: docker/LICENSE (hunk 5 above)
Comment (Babacooll) [reply docker/LICENSE:3]: @jbout should be fixed into develop later

**Comment by Babacooll** [comment]
I'll merge this as this RC has already been QAed as part of the community release. All fixes will be done into develop.

**Pull request merged**